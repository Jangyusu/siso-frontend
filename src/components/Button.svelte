<script lang="ts">
  let buttonText = 'Button';

  const handleClick = () => {
    buttonText = 'Button Clicked';
  };
</script>

<h1>Test button</h1>
<button on:click={handleClick}>{buttonText}</button>
