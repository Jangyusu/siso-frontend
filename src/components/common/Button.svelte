<script lang="ts">
  export let onClick = () => {
    console.log('button clicked');
  };
  export let src = '/';
  export let alt = '';
  export let text: string | number = '';
</script>

<button on:click={onClick} class={$$props.class} style={$$props.style}>
  <img {src} {alt} />
  {text}
</button>

<style lang="scss">
  button {
    padding: 0;
    border: none;
    background: transparent;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 1px;
    font-size: 11px;
    color: $White;
    &.circle {
      border-radius: 100%;
    }
    &.size-lg {
      width: 48px;
      height: 48px;
    }
    &.size-md {
      width: 36px;
      height: 36px;
    }
    &.size-sm {
      width: 24px;
      height: 24px;
    }
    &.img-sm {
      img {
        width: 24px;
        height: 24px;
      }
    }
    &.img-md {
      img {
        width: 36px;
        height: 36px;
      }
    }
  }

  img {
    width: 100%;
  }
</style>
