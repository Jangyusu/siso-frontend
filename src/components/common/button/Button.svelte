<script lang="ts">
  export let onClick = () => {
    console.log('button clicked');
  };
  export let src = '/';
  export let alt = '';
  export let text: string | number = '';
  export let size = 'md';
  export let imgSize = 'md';
</script>

<button
  on:click={onClick}
  class={`${size === 'lg' ? 'w-12 h-12' : 'w-9 h-9'} `}
  style={$$props.style}
>
  <img class={`${imgSize === 'lg' ? 'w-9' : 'w-6'} w-full`} {src} {alt} />
  {text}
</button>

<style lang="postcss">
  button {
    @apply flex flex-col justify-center items-center p-0 bg-transparent text-xs text-white border-0 gap-[1px];
  }
</style>
