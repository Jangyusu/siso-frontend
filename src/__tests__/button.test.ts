import '@testing-library/jest-dom';
import {render} from '@testing-library/svelte';

import Button from '$components/common/button/Button.svelte';

describe('Button', () => {
  const src = '/';
  const alt = 'alt';
  const text = 'icon button';
  const size = 'md';
  const imgSize = 'md';

  test('should match snapshot', () => {
    const {container} = render(Button, {
      props: {src, alt, text, size, imgSize}
    });
    expect(container).toMatchSnapshot();
  });

  test('should have props', () => {
    const {getByText} = render(Button, {
      props: {src, alt, text, size, imgSize}
    });
    expect(getByText(text)).toBeInTheDocument();
  });
});
