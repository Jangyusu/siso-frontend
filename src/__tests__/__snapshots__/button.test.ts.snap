// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<body>
  <div>
    <button
      class="w-9 h-9  svelte-1o342qa"
    >
      <img
        alt="alt"
        class="w-6 w-full"
        src="/"
      />
       
      icon button
    </button>
  </div>
</body>
`;
